test/processStates_test.js

```diff
@@ -0,0 +1,260 @@
+goog.module('datalayerhelper.helper.testing.processStates');
+goog.setTestOnly();
+
+const {DataLayerHelper, buildAbstractModelInterface_} = goog.require('helper');
+
+describe('The processStates function', () => {
+  /**
+   * Asserts that calling processStates_ with the given arguments will result
+   * in the expected model state and expected calls to the listener.
+   *
+   * @param {!Array<Object>} states The states argument for the call to
```

Comment:
!Array<!Object<*>>

Reply:
Sorry, I added the exclamation mark last time and then my brain turned off... I'll be more careful writing JSdoc in the future, to treat it more like code and not comments

Reply:
I see. This makes sense. Thank you!